gobblin-utility/src/main/java/gobblin/util/ConfigUtils.java

```diff
@@ -149,16 +188,33 @@ public static Config propertiesToConfig(Properties properties, Optional<String>
       blacklistedKeys = new HashSet<>(Splitter.on(',').omitEmptyStrings().trimResults()
           .splitToList(properties.getProperty(GOBBLIN_CONFIG_BLACKLIST_KEYS)));
     }
+
+    Set<String> fullPrefixKeys = findFullPrefixKeys(properties, prefix);
+
     ImmutableMap.Builder<String, Object> immutableMapBuilder = ImmutableMap.builder();
     for (Map.Entry<Object, Object> entry : properties.entrySet()) {
       String entryKey = entry.getKey().toString();
-      if (StringUtils.startsWith(entryKey, prefix.or(StringUtils.EMPTY)) && !blacklistedKeys.contains(entryKey)) {
+      if (StringUtils.startsWith(entryKey, prefix.or(StringUtils.EMPTY)) &&
+          !blacklistedKeys.contains(entryKey)) {
+        if (fullPrefixKeys.contains(entryKey)) {
```

Comment:
This should never happen but in a rare case where user's key already has a substring `.ROOT_VALUE` we may want to throw some exception saying `.ROOT_VALUE` is a keyword